### PR TITLE
Proof of concept: render a Gutenberg component in Calypso DevDocs

### PR DESCRIPTION
**Pull request by mmtr**
### DO NOT MERGE
Work in progress

### Summary
The next step of the current work for integrating the Gutenberg blocks in Calypso is to develop a proof on concept prototyping what a single Gutenberg block should look like in the new "Gutenberg Blocks" section in the DevDocs.

As the Gutenberg `core-blocks` package is not ready yet (WordPress/gutenberg#3955), this version of the POC will use the `@wordpress/components` package in order to move forward and identify potential blockers.

#### Notes
- ~~Built on top of #26065, so this PR contains files that have actually modified there.~~ #26064 merged
- ~~As the npm packages are not published yet, it's required to build them locally. I have included a git submodule for helping that~~ [npm package published](https://www.npmjs.com/package/@wordpress/components)!

### Tasks
- [x] Declare `@wordpress/components` as  dependency
- [x] Add documentation for the `Button` Gutenberg component
- [x] Include the `Button` Gutenberg component in the Playground section
- [x] Check there are no issues with the `@wordpress/compose`, `@wordpress/deprecated` and `@wordpress/dom` sub deps.

### Instructions
- `npm install`
- `npm start`
- In a browser, go to the ["Gutenberg blocks" section](http://calypso.localhost:3000/devdocs/gutenberg-blocks)
- The `Button` component from Gutenberg should be rendered

### Blockers / Problems
- ~~`@wordpress/utils` is still used, so I needed to include a `resolve` in `webpack.config.js` handling it~~ Solved in WordPress/gutenberg#7640

screen shot 2018-07-16 at 17 27 40

- ~~There is a SCSS file in one of the Gutenberg components (`scroll-lock`) that doesn't match the expected pattern for the styles (it should be `style.scss` instead of `index.scss` and it should be imported in `packages/components/src/styles.scss` instead of in `packages/components/src/scroll-block/index.js`). This is causing an error I have not been able to solve with any work-around, as I think it should be fixed on Gutenberg~~ Solved in WordPress/gutenberg#7640

screen shot 2018-07-16 at 17 27 57

- ~~Styles should be included either via a CSS loader in order to avoid the warning below.~~ It will be solved with the new `node-sass` version (sass/node-sass#2362)

screen shot 2018-07-16 at 19 41 57

- ~~`@babel/runtime@7.0.0-beta.52` is required by the `@wordpress/components`. This can be be fixed either by updating Babel in Calypso or by declaring a `@babel/runtine` dependency in `@wordpress/components`. Any project using `@wordpress/components` will have the same issue if `@babel/runtime` is not installed or if the installed version doesn’t match the one used during the package build process. It should be added to the `@wordpress/components` package if we don't want to rely on the installed dependencies on a project using it, so the package works out-of-the-box.~~ Solved in WordPress/gutenberg#8057

### CODE REVIEW

**Comment by gziolo** [comment]
`@wordpress/components` shouldn't depend on `@wordpress/utils` anymore. I merged 2 PRs which contributed to that earlier today. You just need to rebase the Gutenberg branch against the latest master :)

**Comment by gziolo** [comment]
> - There is a SCSS file in one of the Gutenberg components (`scroll-lock`) that doesn't match the expected pattern for the styles (it should be `style.scss` instead of `index.scss` and it should be imported in `packages/components/src/styles.scss` instead of in `packages/components/src/scroll-block/index.js`). This is causing an error I have not been able to solve with any work-around, as I think it should be fixed on Gutenberg
> 
> screen shot 2018-07-16 at 17 27 57

@youknowriad, any ideas why it happens?

**Comment by youknowriad** [comment]
> *  There is a SCSS file in one of the Gutenberg components (scroll-lock) that doesn't match the expected pattern for the styles (it should be style.scss instead of index.scss and it should be imported in packages/components/src/styles.scss instead of in packages/components/src/scroll-block/index.js). This is causing an error I have not been able to solve with any work-around, as I think it should be fixed on Gutenberg

@mmtr @gziolo  Good catch, the https://github.com/WordPress/gutenberg/pull/7640 has been updated and should be ready for a final review before merging which hopefully will simplify the work being done here.

**Comment by mmtr** [comment]
Thanks both @gziolo @youknowriad! I'm no longer getting those errors :)

**Comment by mmtr** [comment]
@gziolo I'm trying to compile the `@wordpress/components` SCSS files in order to customize the styles using the SCSS variables defined in Calypso, but I'm getting this error:

screen shot 2018-07-17 at 11 55 25

Not sure why the `react-datepicker` package is not in the `node_modules` of `@wordpress/components` despite of being defined as dependency in the `package.json` file. 

screen shot 2018-07-17 at 12 08 09

Did I skip any step for building the packages locally?

**Comment by youknowriad** [comment]
>  I'm trying to compile the @wordpress/components SCSS files in order to customize the styles using the SCSS variables defined in Calypso, but I'm getting this error:

While I understand the reasoning behind this change, I have some concerns about it. `Sass` files are not considered deliverables in our npm modules. Gutenberg is using them today but once we ship npm modules, I doubt you'll have access to them. Also, it's just an implementation detail, we could consider switching to CSS in JS for example in which case the Sass files won't exist anymore.

I tend to think we should reconsider this and only override styles with CSS. Sass files and sass variables are not an officially supported API in our npm packages.

**Comment by mmtr** [comment]
Thanks @youknowriad, I see your point. 

But that would depend on how we want to deliver the `@wordpress/components` package. Do we want to offer flexibility so anyone can easily customize them in order to match a specific visual identity? If yes, I would reconsider to provide some mechanisms for doing that. If not, the current CSS approach will be enough. I guess this flexibility feature is not a high priority now.

So the question now is, do we want to customize the Gutenberg components in Calypso in order to match the Calypso visual identity or is ok to leave them as originally defined by Gutenberg. What do you think @ehg?

**Comment by gziolo** [comment]
@mmtr, we are merging `@wordpress/components` into Gutenberg `master` later today. We should publish it to npm tomorrow (hopefully - depends on a few cleanup PRs which are in review).

**Comment by mmtr** [comment]
Thanks for the update @gziolo! Do you you have any plan for start working on the `core-blocks` package?

**Comment by gziolo** [comment]
`core-blocks` depends on `components`, `editor` and `blocks`, so it's far on the list :)

**Comment by gziolo** [comment]
For the record, `@wordpress/components` was published to npm and is now referenced in this PR 🎉

**Comment by mmtr** [comment]
I have just updated the description describing a new issue we have related to the `@babel/runtime` package

**Comment by gwwar** [comment]
✨ This also ties in nicely as part of https://github.com/Automattic/wp-calypso/issues/26180

**Comment by gwwar** [comment]
@gziolo @youknowriad 

> @babel/runtime@7.0.0-beta.52 is required by the @wordpress/components

Since components in a library, would it be possible to move dependencies needed to build the js/css bundle to `devDependencies`? Once we publish to npm, the parent package doesn't need to build additional files, right?

File: webpack.config.js [hunk 1]
```diff
@@ -241,10 +241,6 @@ if ( calypsoEnv === 'desktop' ) {
 	// no chunks or dll here, just one big file for the desktop app
 	webpackConfig.output.filename = '[name].js';
 	webpackConfig.output.chunkFilename = '[name].js';
-} else {
-	// jquery is only needed in the build for the desktop app
-	// see electron bug: https://github.com/atom/electron/issues/254
-	webpackConfig.externals.push( 'jquery' );
```
Comment (gwwar) [inline webpack.config.js]: Does `@wordpress/components` still depend on jQuery?

Comment (mmtr) [reply webpack.config.js]: yep. `@wordpress/components` [needs `@wordpress/api-fetch`](https://github.com/WordPress/gutenberg/blob/master/packages/components/package.json#L28), and this one [needs jQuery](https://github.com/WordPress/gutenberg/blob/master/packages/api-fetch/package.json#L24)

Comment (gwwar) [reply webpack.config.js]: Complete aside, but if you type <kbd>y</kbd> on a GitHub file view, it'll give you a permanent link to the file like so: https://github.com/WordPress/gutenberg/blob/348e463b7cbdd246cf0744f6fa1459fefd0f37a7/packages/components/package.json#L28 (so the highlighted line remains the same if folks click on it in the future)

Comment (gwwar) [reply webpack.config.js]: Ah I totally misread the summary of https://github.com/WordPress/gutenberg/pull/7566, they're starting to move away from `jQuery` but it's still a dep

**Comment by gwwar** [comment]
Hmm, taking a quick look, I had a little trouble seeing where the `babel/runtime@7.0.0-beta.52` is coming from, I'll 🔍a bit later, and will make a PR if I find it.

**Comment by gziolo** [comment]
We landed https://github.com/WordPress/gutenberg/pull/8057 and published updated packages to npm. You shouldn't need to provide `babel/runtime` as a dependency anymore. We still should look into how optimize distributed packages as it probably is far from idea 😅

**Comment by mmtr** [comment]
everything looks good so far with the `@wordpress/components` package, so I'm closing this PR as we don't want to expose the Gutenberg components in Calypso for the moment.

the goal of this was to identify potential blockers and I think we're done with that :) 

thanks everyone involved!

**Pull request closed without merging**